modules/layers/src/scatterplot-layer/scatterplot-layer.ts

```diff
@@ -235,6 +238,7 @@ export default class ScatterplotLayer<DataT = any, ExtraPropsT extends {} = {}>
     if (params.changeFlags.extensionsChanged) {
       this.state.model?.destroy();
       this.state.model = this._getModel();
+      this.state.uniformStore = new UniformStore<{picking: any}>({picking});
```

Comment:
Note that you should be able to initialize the UniformStore up in the class declaration.

Reply:
Maybe instead we need a `uniformStore.reset()` here?